src/components/virtualRepeat/virtual-repeater.js

```diff
@@ -279,6 +313,15 @@ VirtualRepeatContainerController.prototype.handleScroll_ = function() {
   this.offsetter.style.webkitTransform = transform;
   this.offsetter.style.transform = transform;
 
+  if(this.bindTopIndex) {
+    var topIndex = Math.floor(offset / itemSize);
+    if(topIndex !== this.topIndex && topIndex < this.repeater.itemsLength) {
+      this.topIndex = topIndex;
+      this.bindTopIndex.assign(this.$scope, topIndex);
+      if (!this.$scope.$root.$$phase) this.$scope.$digest();
```

Comment:
Did some testing... we definitely need safe digest:
`Error: [$rootScope:inprog] $digest already in progress
http://errors.angularjs.org/1.4.6/$rootScope/inprog?p0=%24digest`